ambry-commons/src/main/java/com.github.ambry.commons/BlobId.java

```diff
@@ -328,10 +336,33 @@ public BlobIdType getType() {
   /**
    * @return {@code true} if the blob that this id represents is encrypted. {@code false} otherwise
    */
-  public boolean isEncrypted() {
+  private boolean isEncrypted() {
     return isEncrypted;
   }
 
+  /**
+   * Check if encrypted bit in blobId is set based on original blobId string.
+   * @return {@code true} if encrypted bit in this string is set. {@code false} otherwise
+   */
+  public static boolean isEncrypted(String blobIdString) {
+    if (blobIdString == null) {
+      return false;
+    }
+    DataInputStream stream =
+        new DataInputStream(new ByteBufferInputStream(ByteBuffer.wrap(Base64.decodeBase64(blobIdString))));
+    boolean isEncrypted = false;
+    try {
+      short version = stream.readShort();
+      if (version >= BLOB_ID_V3) {
+        isEncrypted = (stream.readByte() & IS_ENCRYPTED_MASK) != 0;
+      }
+    } catch (IOException e) {
+      isEncrypted = false;
```

Comment:
I don't think this should be caught either. The error should not be silently swallowed, let this propagate outside this method, it is the caller's fault.